app/src/main/java/org/mozilla/fenix/home/collections/Collection.kt

```diff
@@ -0,0 +1,213 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Paint
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.state.recover.RecoverableTab
+import mozilla.components.concept.engine.Engine
+import mozilla.components.feature.tab.collections.Tab
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.R.drawable
+import org.mozilla.fenix.R.string
+import org.mozilla.fenix.compose.list.ExpandableListHeader
+import org.mozilla.fenix.ext.getIconColor
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Rectangular shape with all corners rounded used to display a collapsed collection.
+ */
+private val collapsedCollectionShape = RoundedCornerShape(8.dp)
+
+/**
+ * Rectangular shape with only the top corners rounded used to display an expanded collection with other views
+ * placed immediately below this which can be shown immediately next to it, with no visible separation.
+ */
+private val expandedCollectionShape = RoundedCornerShape(topStart = 8.dp, topEnd = 8.dp)
+
+/**
+ * Composable used to display an individual [TabCollection].
+ *
+ * @param collection [TabCollection] to display.
+ * @param expanded Whether the collection is expanded to show it's containing tabs or not.
+ * @param menuItems List of [CollectionMenuItem] to be shown in a menu.
+ * @param onToggleCollectionExpanded Invoked when the user clicks on the collection.
+ * @param onCollectionShareTabsClicked Invoked when the user clicks to share the collection.
+ * @param onCollectionMenuOpened Invoked when the user clicks to open a menu for the collection.
+ */
+@Composable
+fun Collection(
+    collection: TabCollection,
+    expanded: Boolean,
+    menuItems: List<CollectionMenuItem>,
+    onToggleCollectionExpanded: (TabCollection, Boolean) -> Unit = { _, _ -> },
+    onCollectionShareTabsClicked: (TabCollection) -> Unit = { _ -> },
+    onCollectionMenuOpened: () -> Unit = {},
+) {
+    var isMenuShown by remember(collection) { mutableStateOf(false) }
+    val isExpanded by remember(collection) { mutableStateOf(expanded) }
+
+    Card(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .clickable { onToggleCollectionExpanded(collection, !isExpanded) }
+            .height(48.dp),
+        shape = if (isExpanded) expandedCollectionShape else collapsedCollectionShape,
+        backgroundColor = FirefoxTheme.colors.layer2,
+        elevation = 5.dp, // This needs to match the elevation of the TabInCollectionViewHolder for the shadow.
```

Comment:
There is a matching comment in the new `TabInCollection` composable - https://github.com/mozilla-mobile/fenix/pull/24682/files#diff-fea09fe41897bb0e80f1fc96ae79d95b4fca055d5a548a5500fcb62a01ede580R118. With how problematic the shadows where I feel like it's an important detail. Will update the comment to rather point to `TabInCollection`. Can remov both if you think they aren't helping.

Reply:
I would pick between removing it or referencing `home_item_elevation` since I would assume matching the card elevation between the two would be obvious, but maybe the best solution is just have a constant that is used by both. Looking ahead, were we thinking of joining the 2 ViewHolders in some way once we are done migrating both of them into Compose. For example, would it be possible to have this all under a single Card in a followup after we landed `TabInCollection`?

Reply:
I also suspect that we should check with UX whether or not we should keep this at 5dp since all the other cards on Home is using 6dp, but this can be a followup.

Reply:
| For example, would it be possible to have this all under a single Card in a followup after we landed TabInCollection
In the future I'd hope so. Having all the collection details - header + items in a single Composable will probably help resolve many of the current issues with broken layouts / inconsistent state.
But this will require significant work to refactor not just the UI but the entire data flow so it will probably take some time to happen.
In the meantime maybe just removing that comment is best.